x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/QuerySettings.java

```diff
@@ -31,11 +31,11 @@ public class QuerySettings {
         true,
         false,
         true,
-        "A project routing expression, "
+        "A project routing literal, "
```

Comment:
I'd prefer to keep "expression" here. This is for documentation, and a routing expression is a specific concept in CPS, "literal" is just an implementation detail that shouldn't leak here IMHO.
(I'm referring only to the message string of course, not to the refactoring that uses Literal class instead of Expression; I like that one)